### PR TITLE
ID-1234 ID-2814 [Feat] UI for custom access rules

### PR DESCRIPTION
**Pull request by squallstar**
- https://weboo.atlassian.net/browse/ID-1234
- https://weboo.atlassian.net/browse/ID-2814

image

### CODE REVIEW

**Review by tonytlwu — APPROVED** [review]

File: templates/accessRule.interface.hbs [hunk 1]
```diff
@@ -14,11 +14,15 @@
         </div>
     </a>
   </td>
-  <td class="align-baseline">{{{allow}}}</td>
-  <td class="align-baseline">{{{type}}}</td>
-  <td class="align-baseline">{{{require}}}</td>
-  <td class="align-baseline">{{{exclude}}}</td>
-  <td class="align-baseline">{{apps}}</td>
+  {{#if hasScript}}
+    <td colspan="5" class="align-baseline"><span class="label label-info"><i class="fa fa-code"></i> Custom rule</span> {{name}}</td>
```
Comment (tonytlwu) [inline templates/accessRule.interface.hbs:18]:
```suggestion
    <td colspan="5" class="align-baseline"><span class="label label-default"><i class="fa fa-code"></i> Custom rule</span> {{name}}</td>
```

File: interface.html [hunk 2]
```diff
@@ -290,91 +291,116 @@ <h4 class="modal-title"></h4>
...
           </div>
-          <hr />
-          <div class="row">
-            <div class="col-md-4">
-              <h5>Request data requirements</h5>
-              <p>Specify rules that data must follow.</p>
+          <div data-rule-custom>
+            <div class="row">
+              <div class="col-md-4">
+                <h5>Rule name</h5>
```
Comment (tonytlwu) [inline interface.html:386]:
```suggestion
                <h5>Security rule name</h5>
```

File: interface.html [hunk 3]
```diff
@@ -290,91 +291,116 @@ <h4 class="modal-title"></h4>
...
+                <p>The description for your custom rule. Write something handy to help you remember what this rule is about.</p>
```
Comment (tonytlwu) [inline interface.html:387]:
```suggestion
                <p>Use a name to describe the security rule.</p>
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Comment by squallstar** [comment]
Reimplemented by https://github.com/Fliplet/fliplet-widget-data-source/pull/208

**Pull request closed without merging**